astropy/config/paths.py

```diff
@@ -19,6 +20,46 @@
 ]
 
 
+class _DirectoryType(Enum):
+    CONFIG = auto()
+    CACHE = auto()
+
+
+def _get_dir_path(rootname: str, dirtype: _DirectoryType) -> Path:
+    match dirtype:
+        case _DirectoryType.CONFIG:
+            cls = set_temp_config
+            fallback = "config"
+        case _DirectoryType.CACHE:
+            cls = set_temp_cache
+            fallback = "cache"
+        case _:
+            raise ValueError(f"Unexpected {dirtype=}")
+
+    # If using set_temp_x, that overrides all
+    if cls._temp_path is not None:
+        xch = cls._temp_path
+        path = xch / rootname
+        if not path.is_file():
+            path.mkdir(exist_ok=True)
+        return path.resolve()
+
+    # first look for XDG_CONFIG_HOME
+    if (
+        (xdg_config_home := os.getenv("XDG_CONFIG_HOME")) is not None
+        and (xch := Path(xdg_config_home)).exists()
+        and not (xchpth := xch / rootname).is_symlink()
+    ):
+        if xchpth.exists():
+            return xchpth.resolve()
+        linkto = xchpth
+    else:
+        # symlink will be set to this if the directory is created
+        linkto = None
```

Comment:
I refactored the logic around binding `linkto` for clarity. Now both logical branches are next to each other and at the same level of indentation, but the the logic is actually strictly equivalent

Reply:
I like the refactor, but shouldn't the comment go up, ahead of l.55?

Reply:
done